### PR TITLE
bash-language-server: fix broken package

### PR DESCRIPTION
**Pull request by ck3d**
## Description of changes

Fix the build phase of the package and add minor changes like, version test, remove unused variable and make shellcheck as default.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ba/bash-language-server/package.nix [hunk 1]
```diff
@@ -53,13 +53,15 @@ stdenv.mkDerivation (finalAttrs: {
 
     # Create the executable, based upon what happens in npmHooks.npmInstallHook
     makeWrapper ${lib.getExe nodejs} $out/bin/bash-language-server \
-      --prefix PATH : ${lib.makeBinPath [ shellcheck ]} \
+      --suffix PATH : ${lib.makeBinPath [ shellcheck ]} \
```
Comment (AndersonTorres) [inline pkgs/by-name/ba/bash-language-server/package.nix:56]: Is this a thing expected or recommended?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by amarshall — APPROVED** [review]
Can confirm package was broken, and that this remedies it. Changes overall look good, but not familiar with this drv specifically.

File: pkgs/by-name/ba/bash-language-server/package.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/by-name/ba/bash-language-server/package.nix:56]: Understood.

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/by-name/ba/bash-language-server/package.nix [hunk 2]
```diff
@@ -60,6 +61,8 @@ stdenv.mkDerivation (finalAttrs: {
     runHook postInstall
   '';
 
+  passthru.tests.version = testers.testVersion { package = finalAttrs.finalPackage; };
```
Comment (doronbehar) [inline pkgs/by-name/ba/bash-language-server/package.nix:63]: Is there a chance you could switch to [`versionCheckHook`](https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook) please?

File: pkgs/by-name/ba/bash-language-server/package.nix [hunk 3]
```diff
@@ -19,9 +19,8 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-yJ81oGd9aNsWQMLvDSgMVVH1//Mw/SVFYFIPsJTQYzE=";
   };
 
-  pnpmWorkspace = "bash-language-server";
   pnpmDeps = pnpm_8.fetchDeps {
-    inherit (finalAttrs) pname version src pnpmWorkspace;
```
Comment (doronbehar) [inline pkgs/by-name/ba/bash-language-server/package.nix:23]: An explanation in the commit message for this change would be nice. `pnpmWorkspace` is an attribute added to support here:

https://github.com/NixOS/nixpkgs/pull/323493

File: pkgs/by-name/ba/bash-language-server/package.nix (hunk 1 above)
Comment (doronbehar) [reply pkgs/by-name/ba/bash-language-server/package.nix:56]: > I think it’s reasonable that if the user has shellcheck in their $PATH, that one should take precedence.

That's indeed a good reason to use `--suffix`. Could you explain that in the commit message please?

**Review by doronbehar — CHANGES_REQUESTED** [review]
I don't understand what this PR is for (besides the `--suffix` change), `bash-language-server` builds fine for me on branch `master`.

File: pkgs/by-name/ba/bash-language-server/package.nix (hunk 3 above)
Comment (ck3d) [reply pkgs/by-name/ba/bash-language-server/package.nix]: I remove the commit. I think this should be discussed in your mentioned PR.

File: pkgs/by-name/ba/bash-language-server/package.nix (hunk 2 above)
Comment (ck3d) [reply pkgs/by-name/ba/bash-language-server/package.nix]: Yes, I could changed to your mentioned hook. Thanks, the hook is new for me.

**Comment by ck3d** [comment]
Without this this PR, the package is built, but the executable fails to start. I incorporated your recommendations.

**Comment by doronbehar** [comment]
> Without this this PR, the package is built, but the executable fails to start. I incorporated your recommendations.

OK I see now what you are talking about. How did you manage to find out the `pnpm compile server` command? I wonder if the documentation for pnpm could be improved...

**Review by doronbehar — APPROVED** [review]

**Comment by doronbehar** [comment]
I will merge when CI is green..

**Comment by ck3d** [comment]
`pnpm compile` is a shorthand for `pnpm run compile`, see https://pnpm.io/8.x/cli/run
Script `compile` is the documented target, see https://github.com/bash-lsp/bash-language-server/blob/main/docs/development-guide.md#initial-setup

I additionally passed `server` as argument to compile to only compile server and not vscode-client. The resulting command is `tsc -b server`.
See also https://github.com/bash-lsp/bash-language-server/blob/main/package.json#L6

**Comment by doronbehar** [comment]
> `pnpm compile` is a shorthand for `pnpm run compile`, see [pnpm.io/8.x/cli/run](https://pnpm.io/8.x/cli/run) Script `compile` is the documented target, see [bash-lsp/bash-language-server@`main`/docs/development-guide.md#initial-setup](https://github.com/bash-lsp/bash-language-server/blob/main/docs/development-guide.md?rgh-link-date=2024-08-10T22%3A04%3A26Z#initial-setup)
> 
> I additionally passed `server` as argument to compile to only compile server and not vscode-client. The resulting command is `tsc -b server`. See also [bash-lsp/bash-language-server@`main`/package.json#L6](https://github.com/bash-lsp/bash-language-server/blob/main/package.json?rgh-link-date=2024-08-10T22%3A04%3A26Z#L6)

Thanks for explaining. I added a note to the main comment of https://github.com/NixOS/nixpkgs/issues/317927 .

**Pull request merged**